### DISCUSSION TITLE
LME partially installed!

### DISCUSSION THREAD

**Discussion by murphypa271327** [cisagov/LME#80] [FAQ] [answered]
Does anyone have a fix for the fact that when deploy.sh runs, it never finishes completely! I was able to get to the website, but no ID and passwords because it never got that far! Is there a file I can edit or a fix somewhere that will let it complete the process?

**Comment by murphypa271327** [comment]
To better explain - this never happens!!!

The usernames and passwords will be provided in a message similar to below.
 
```
##################################################################################
## Kibana/Elasticsearch Credentials are (these will not be accessible again!!!!) ##
##
## Web Interface login:
## elastic:<PASSWORD>
##
## System Credentials
## kibana:<PASSWORD>
## logstash_system:<PASSWORD>
## logstash_writer:<PASSWORD>
## dashboard_update:<PASSWORD>
##################################################################################

**Comment by murphypa271327** [comment]
Now when I run the deploy.sh it stops and displays the message Elasticsearch is not responding after 180 attempts - exiting

**Comment by llwaterhouse** [comment]
Please go through troubleshooting.md.  It describes fixes for the most common issues.

**Reply by llwaterhouse** [reply]
In particular, this section might be of interest: https://github.com/cisagov/LME/blob/main/docs/markdown/reference/troubleshooting.md#deploysh-stalls-on-wating-for-elasticsearch-to-connect

**Comment by murphypa271327** [comment]
Good Afternoon,
Thank you for responding, I am by know means a linux guy! LOL I’ve been working Windows for the past 30+ years, this is the first real demand for me to work on linux here, but it’s Redhat! If would be so kind as too look at this and maybe shed some light! I never get to the point where it gives me the passwords!

[X] Waiting for Elasticsearch to be ready
.......................................................................................................Elasticsearch is up and running.
[X] Setting elastic user password
{}
[X] Setting kibana system password
{}
[X] Setting logstash system password
{}
[X] Setting logstash writer role
{"role":{"created":true}}
[X] Setting dashboard update role
{"role":{"created":true}}
[X] Creating logstash writer user
{"created":true}
[X] Setting logstash writer password
{}
[X] Creating dashboard update user
{"created":true}
[X] Setting dashboard update user password
{}
[X] Configuring elasticsearch Replica settings
{"error":{"root_cause":[{"type":"parse_exception","reason":"unknown key [template] in the template "}],"type":"parse_exception","reason":"unknown key [template] in the template "},"status":400}{"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [[_all]]","index_uuid":"_na_","index":"[_all]"}],"type":"index_not_found_exception","reason":"no such index [[_all]]","index_uuid":"_na_","index":"[_all]"},"status":404}
[X] Generating files_for_windows zip
  adding: tmp/lme/ (stored 0%)
  adding: tmp/lme/winlogbeat.yml (deflated 52%)
  adding: tmp/lme/wlbclient.key (deflated 24%)
  adding: tmp/lme/wlbclient.crt (deflated 24%)
  adding: tmp/lme/root-ca.crt (deflated 25%)
test of /opt/lme/files_for_windows.zip OK

[X] Setting Elastic pipelines
{"acknowledged":true}[X] We think your main disk is 98G
[!] Unable to determine retention policy - exiting
***@***.***:/opt/lme/Chapter 3 Files#


Here is the screen I get! Almost there! LOL
***@***.***

This is starting with a fresh install of Ubuntu 22.04 – I tried the removal and reinstall! And I got this!
I could uninstall it again and this time also try the removal of Docker. Was just hoping you might know prior to me doing that!

Thank you so much


***@***.***Patrick Murphy
Network Systems Analyst
Office of Information Technology
301 S. Monroe Street
Tallahassee, FL  32301
ph: (850) 606-5560 / fax: (850) 606-5501
                              ***@***.******@***.***>

**Comment by llwaterhouse** [comment]
LME supports only Ubuntu Linux at this time, not Redhat.  You mention Redhat at the top, but then refer to Ubuntu in the middle of your message so it is unclear which Linux you are using.

Often the quickest way to get things to work is to uninstall and re-start installation.  If you do uninstall everything, be sure to remove your docker volumes after:
```
sudo docker volume rm lme_esdata
sudo docker volume rm lme_logstashdata
```
Other people have had issues with using 98 Gb (although in our environment that is enough).  You could try increasing your disk space to 128Gb.

**Comment by llwaterhouse** [comment]
On your Linux server, please try running 
`df -h -l -t ext4 --output=source,size /var/lib/docker`

You should get output that looks similar to this
![image](https://github.com/cisagov/LME/assets/82845774/a3253a55-a4a4-49e6-a66f-4046e6d33de6)

If you get output, we suggest you  increase the size of your disk to 128Gb.

 If it doesn't work, then there is some other problem so please  look through troubleshooting.md for answers.

**Comment by murphypa271327** [comment, marked as answer]
OK I had 200GB, but it only assigned itself the 98GB. I extended that to the full 200GB, Then I did the following and it's now working. I just have to get the windows systems communicating! LOL
Here are my steps:
# Some of the steps are just because and to play it safe!!!

#Remove LME, so you can start from scratch.
cd /opt/lme/Chapter\ 3\ Files/
sudo ./deploy.sh uninstall
rm -r /opt/lme

# Completely Uninstall Docker.
dpkg -l | grep -i docker
sudo apt-get purge -y docker-engine docker docker.io docker-ce docker-ce-cli docker-compose-plugin
sudo apt-get autoremove -y --purge docker-engine docker docker.io docker-ce docker-compose-plugin
sudo rm -rf /var/lib/docker /etc/docker
sudo rm /etc/apparmor.d/docker
sudo groupdel docker
sudo rm -rf /var/run/docker.sock
sudo docker volume rm lme_esdata
sudo docker volume rm lme_logstashdata

#Restart your system
sudo reboot

#Reinstall LME
sudo apt update
sudo apt install git -y
sudo git clone https://github.com/cisagov/lme.git /opt/lme/
Cd /opt/lme/"Chapter 3 Files"
sudo ./deploy.sh install

This worked for me! Thank you for the help! Now I can finish the setup! Whew!!!

**Comment by llwaterhouse** [comment]
Glad you got it to work!
